homeassistant/components/homewizard/button.py

```diff
@@ -54,4 +57,7 @@ def __init__(
 
     async def async_press(self) -> None:
         """Identify the device."""
-        await self.coordinator.api.identify()
+        try:
+            await self.coordinator.api.identify()
+        except RequestError as ex:
+            raise HomeAssistantError(str(ex)) from ex
```

Comment:
Exceptions like these still throw a big error in the log, is this intentional or did I implement this incorrectly?

Reply:
Also, there is no visual feedback for the user such as 'unavailable'

Reply:
> Exceptions like these still throw a big error in the log

Yes, they make a trace in the logs, which is helpful when looking for issues. The `HomeAssistantError` makes it possible to ignore them (and also properly bubble up the message to the UI).

If you are not providing an custom message (because I assume the message is already user friendly?)

```suggestion
        try:
            await self.coordinator.api.identify()
        except RequestError as ex:
            raise HomeAssistantError from ex
```

> Also, there is no visual feedback for the user such as 'unavailable'

I'm not sure if I follow that?

Reply:
> Also, there is no visual feedback for the user such as 'unavailable'

When I force an error (disabled polling, turn off device) there was no bubble or frontend error message, the switch just flipped back after the timeout. I assume it just took to long for HA to handle the error?

Tried it with a 'Disabled API'. This indeed works as expected; a nice popup appears.

Reply:
`unavailable` are internally skipped, and will indeed not cause errors. This is by design. Although, it looks weird in some places (I agree).